tests/integration_tests/functional/test_cpu_features.py

```diff
@@ -612,7 +619,7 @@ def _test_cpu_wrmsr_restore(context):
     tmp_snapshot_artifacts_dir = (
         Path() / chroot_dir / "tmp" / context.kernel.base_name()
     )
-    os.makedirs(tmp_snapshot_artifacts_dir)
+    clean_and_mkdir(tmp_snapshot_artifacts_dir)
```

Comment:
Could you explain a bit how it works if run locally (both snapshot and restore on the same machine). How are the artifacts transferred between the two parts? Is this not removing them before they are consumed?

Reply:
The snapshots created the path specified by snapshot_artifacts_dir ("tests/snapshot_artifacts/wrmsr/").
tmp_snapshot_artifacts_dir (e.g. "/srv/jailer/firecracker/d16e2e7e-99b8-4519-8975-6f593b507801/root/tmp/vmlinux-4.14 ") is the temporary jailer path.
Files are copied from snapshot_artifacts_dir to tmp_snapshot_artifacts_dir and then used to restore vm.

So we are not removing the snapshot files before consuming them, we are just making sure the temp jailer directory is clean when we copy the artifacts to it. The same is done in the _test_cpu_cpuid_restore().

Also, the new code just replaces os.makedirs() with shutil.rmtree()+os.makedirs() which doesn't change the existing working behaviour. If a directory already existed then the old code would have crashed at os.makedirs(tmp_snapshot_artifacts_dir) anyways.

Reply:
Clear, thank you!